lib/rules/components.ts

```diff
@@ -1,67 +1,70 @@
-import { RuleBase, IRule } from "./rules-base";
 import { DiagnosticSeverity } from "vscode-languageserver";
-import { OFile, OPackage } from "../parser/objects";
+import { OComponent, OFile, OPackage } from "../parser/objects";
+import { IRule, RuleBase } from "./rules-base";
 
 export class RComponent extends RuleBase implements IRule {
   public static readonly ruleName = 'component';
   file: OFile;
 
   check() {
     for (const architecture of [...this.file.architectures, ...this.file.packages.filter(p => p instanceof OPackage) as OPackage[]]) {
```

Comment:
This seems awkward, what if a compoment declaration happens in a declarativePart of a block or generate?